server/src/test/java/org/elasticsearch/client/internal/OriginSettingClientTests.java

```diff
@@ -37,7 +37,6 @@ protected <Request extends ActionRequest, Response extends ActionResponse> void
                     ActionListener<Response> listener
                 ) {
                     assertEquals(origin, threadPool().getThreadContext().getTransient(ThreadContext.ACTION_ORIGIN_TRANSIENT_NAME));
-                    super.doExecute(action, request, listener);
```

Comment:
Neither here nor in the other test do we care about resolving the listener. The result passed by the noop client is always `null`. I figured this was still better than adding a null check or hackily casting a non-null response in the  noop-client ...